grails-app/services/org/pih/warehouse/tableroapi/IndicatorDataService.groovy

```diff
@@ -110,46 +110,40 @@ class IndicatorDataService {
         return indicatorData;
     }
 
-DataGraph getSentStockMovements(def location, def params) {
-    Integer querySize = params.querySize? params.querySize.toInteger() : 5
-    List allLocations = dataService.executeQuery("select id from location")
-    List listDatasets = []
-    List listLabel = []
-    today.clearTime()
-    // Query data for all locations
-    for(item in allLocations) {
-        List listData = []
-        listLabel = []
-        try {
-            List locationName = dataService.executeQuery("select name from location where id=" + item[0].value)
-            // querySize is the quantity of months in the filter : until which month query data
+    DataGraph getSentStockMovements(def location, def params) {
+        Integer querySize = params.querySize? params.querySize.toInteger() : 5
+        today.clearTime()
+
+        String query = "SELECT COUNT(s.id) as total, YEAR(s.last_updated) as year, s.destination_id as destination, MONTH(s.last_updated) as month FROM openboxes.shipment s WHERE origin_id = '" + location.id + "' AND s.current_status <> 'PENDING' GROUP BY MONTH(s.last_updated), s.destination_id, YEAR(s.last_updated)"
+
+        List response = dataService.executeQuery(query)
+        List listRes = []
+        List listLabel = []
+        for(item in response) {
+            Location itemLocation = Location.get(item.destination)
+            List listData = []
+            listLabel = []
+
             for(int i = querySize; i >= 0; i--) {
-                def monthBegin = today.clone()
-                def monthEnd = today.clone()
-                monthBegin.set(month: today.month - i, date: 1)
-                monthEnd.set(month: today.month - i + 1, date: 1)
+                Date month = today.clone()
+                month.set(month: today.month - i, date: 1)
 
-                def query = Shipment.executeQuery("select count(*) from Shipment s where s.lastUpdated >= :monthOne and s.lastUpdated < :monthTwo and s.origin = " + item[0].value + " and s.currentStatus <> 'PENDING'", 
-                // The column transaction_transaction_date doesn't exist, using lastUpdated instead
-                ['monthOne': monthBegin, 'monthTwo': monthEnd]);
-                String monthLabel = new java.text.DateFormatSymbols().months[monthBegin.month].substring(0,3)
+                Integer value = 0
+                if (month.month == item.month-1 && month.year + 1900 == item.year) {
```

Comment:
Please add a comment. Depending on what you're trying to do I might suggest another approach.

Reply:
In fact, can you comment on all of this ... the two loops as well.